### PR TITLE
Testing level failures

### PR DESCRIPTION
**Pull request by faucomte**
Added tests which make sure blockly levels fail when:
- the van runs out of fuel
- the van runs out of instructions
- the van runs a red light
- the van doesn't deliver to every destination
- a procedure is undefined in blockly

### CODE REVIEW

**Comment by faucomte** [comment]
**Pull request summary:**
File test_crashes has been renamed to test_level_failures!
The following tests have been added:

- check that the 'running out of fuel' message pops up when the van runs out of fuel.
- check that the 'running out of instructions' message pops up when the van runs out of instructions.
- check that the 'not delivered everywhere' message pops up when the van doesn't deliver to all destinations.
- check that the 'program isn't quite right' message pops up when a procedure is undefined in blockly.

**Tests runtime:**
12 tests run in 1.5 minutes. 

**Coverage:**
Coverage increased ever so slightly by 0.01%.

**Comment by CelineBoudier** [comment]
@faucomte please rebase :)

**Comment by duncanrussell-ocado** [comment]
need to fix conflicts before merge...

**Comment by faucomte** [comment]
Rebasing...

**Comment by faucomte** [comment]
@duncanrussell-ocado @ChrisMukherjee @CelineBoudier All rebased!

File: game/end_to_end_tests/game_page.py [hunk 1]
```diff
@@ -151,9 +151,21 @@ def run_crashing_program(self):
...
+    def run_out_of_instructions_program(self):
         return self._run_failing_program("The van ran out of instructions before it reached a destination.")
 
+    def run_out_of_fuel_program(self):
+        return self._run_failing_program("You ran out of fuel!")
+
+    def run_a_red_light_program(self):
+        return self._run_failing_program("Uh oh, you just sent the van through a red light!")
+
+    def run_program_that_does_not_deliver_everywhere(self):
```
Comment (duncanrussell-ocado) [inline game/end_to_end_tests/game_page.py]: naming convention... please match other run_x_program() i.e. run_does_not_deliver_everywhere_program()

**Pull request merged**